openmmtools/integrators.py

```diff
@@ -1093,14 +1098,31 @@ def substep_function(step_string):
 
         if measure_shadow_work:
             self.addGlobalVariable("old_pe", 0)
-            self.addGlobalVariable("new_pe", 0)
+
             self.addGlobalVariable("shadow_work", 0)
 
+        if measure_protocol_work or measure_shadow_work:
+            self.addGlobalVariable("new_pe", 0)
+
+        if measure_protocol_work:
+            self.addGlobalVariable("protocol_work", 0)
+            self.addGlobalVariable("perturbed_pe", 0)
+
         # Integrate
         self.addUpdateContextState()
         self.addComputeTemperatureDependentConstants({"sigma": "sqrt(kT/m)"})
+
+        # Protocol work is calculated by taking the potential energy in the perturbed system
+        # and subtracting the previous, unperturbed potential energy
+        # before any steps are taken.
+        if measure_protocol_work:
+            self.addComputeGlobal("perturbed_pe", "energy")
+            self.addComputeGlobal("protocol_work", "protocol_work + (perturbed_pe - new_pe)")
```

Comment:
I find it difficult to wrap my head around `CustomIntegrators` at the best of times, but is there potential for conflict if `measure_shadow_work = True`, as `new_pe` is set to zero?